src/Statement/StatementListHolder.php

```diff
@@ -6,6 +6,7 @@
  * Interface for classes that contain a StatementList.
  *
  * @since 3.0
+ * @deprecated since 5.1
```

Comment:
I agree. That means that 6.0 will be breaking without a smooth migration path. You either migrate your code to 6.0, your you don't. Which is fine - that's what major releases are for.